pebble_tool/util/logs.py

```diff
@@ -55,8 +57,41 @@ def __init__(self, pebble, force_colour=None):
                                                                self.handle_phone_log))
         self.handles.append(pebble.register_transport_endpoint(MessageTargetPhone, WebSocketConnectionStatusUpdate,
                                                                self.handle_connection))
+        self.sourcemap = self._load_js_sourcemap()
         add_tools_to_path()
 
+    def _load_js_sourcemap(self):
+        sourcemap_path = "build/pebble-js-app.js.map"
+        if not os.path.exists(sourcemap_path):
+            return None
+        with open(sourcemap_path) as sourcemap_file:
+            try:
+                return sourcemap.load(sourcemap_file)
+            except SourceMapDecodeError as e:
+                logger.warning('Found %s, but failed to parse it: %s', sourcemap_path, str(e))
+                return None
+
+    def _sourcemap_translate_js_log(self, logstr):
+        if not self.sourcemap:
+            return logstr
+
+        def sourcemap_replacer(matchobj):
+            d = matchobj.groupdict()
+            line = int(d['line'])
+            column = int(d['column']) if 'column' in d else 0
+            try:
+                token = self.sourcemap.lookup(line - 1, column)  # sourcemap wants zero-based lines numbers!
+                return u"{}:{}:{}".format(token.src, token.src_line + 1, token.src_col)
+            except IndexError:
+                return u"???:?:?"
```

Comment:
For certain lines/cols the `sourcemap` module craps out.
Not sure why. I cross-checked with Mozilla's source-map .js module and it also craps out, so perhaps the webpack source map generation is slightly broken?